### PR TITLE
Added event log documentation

### PR DESCRIPTION
**Pull request by trmartin4**
## Objective

Added documentation of Event Logs.

### CODE REVIEW

**Review by Hinton — COMMENTED** [review]
Some personal thoughts on how we can streamline this slightly. I feel like we're jumping between different knowledge expectations. Let me know if you disagree on any of the comments as they are mostly personal preferences.

File: docs/architecture/event-logs/index.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+---
+
+# Event Logs
+
+Bitwarden event logs are used for Teams and Enterprise organizations to capture timestamped records
+of events that occur within the organization.
+
+For documentation on how to view events, see our Help documentation
+[here](https://bitwarden.com/help/event-logs/).
+
```
[the remark this answers is not in the document]
Comment (trmartin4) [reply docs/architecture/event-logs/index.md:24]: That's a great clarification.  I was focused on client-side events, as they are more complicated, but server-side need to be highlighed as well.

File: docs/architecture/event-logs/index.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+When uploaded, event logs are sent to the server through `POST` requests to the `/collect` endpoint
+on the `Events` service.
+
+At this point, the handling of the logs differs based on the hosting configuration of the Bitwarden
+instance. This configuration is done through Dependency Injection in the
+[`Startup`](https://github.com/bitwarden/server/blob/master/src/Events/Startup.cs) class on the
+`Events` project. Namely, the implementation of
+[`IEventWriteService`](https://github.com/bitwarden/server/blob/master/src/Core/Services/IEventWriteService.cs)
+differs based on whether the instance is self-hosted or cloud-hosted.
```
[the remark this answers is not in the document]
Comment (trmartin4) [reply docs/architecture/event-logs/index.md]: I've re-worded this quite a bit to follow your recommendation.

**Comment by trmartin4** [comment]
@Hinton, thank you for the review.  I think it reads much more cleanly now without all of the references to specific DI implementations.  Just specifying the interface name should give someone the right direction once they're in an IDE.

As for the target audience, I was targeting mainly a potential community contributor or, more likely, a Bitwarden engineer who has to make changes or additions to the events.  We had to add some events for some work on the SG pod and there was confusion around whether we were dealing with `ReferenceEvents` or `Events`, and how all the different services worked together for event logs.

**Review by Hinton — APPROVED** [review]

**Pull request merged**